ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/subscription/logs/LogsSubscriptionService.java

```diff
@@ -46,9 +54,34 @@ public void accept(final LogWithMetadata logWithMetadata) {
                   && filterParameter.getToBlock().getNumber().orElse(Long.MAX_VALUE) >= blockNumber
                   && filterParameter.getLogsQuery().matches(logWithMetadata);
             })
-        .forEach(
-            logsSubscription ->
-                subscriptionManager.sendMessage(
-                    logsSubscription.getSubscriptionId(), new LogResult(logWithMetadata)));
+        .forEach(logsSubscription -> sendLogToSubscription(logWithMetadata, logsSubscription));
+  }
+
+  public void checkPrivateLogs(final BlockAddedEvent event) {
+    privacyQueries.ifPresent(
+        pq ->
+            subscriptionManager.subscriptionsOfType(SubscriptionType.LOGS, LogsSubscription.class)
+                .stream()
+                .filter(PrivateLogsSubscription.class::isInstance)
+                .map(PrivateLogsSubscription.class::cast)
+                .forEach(queryPrivateEventForSubscription(pq, event)));
+  }
+
+  private Consumer<PrivateLogsSubscription> queryPrivateEventForSubscription(
+      final PrivacyQueries privacyQueries, final BlockAddedEvent event) {
+    return subscription -> {
+      final String privacyGroupId = subscription.getPrivacyGroupId();
+      final LogsQuery logsQuery = subscription.getFilterParameter().getLogsQuery();
+
+      privacyQueries
+          .matchingLogs(privacyGroupId, event.getBlock().getHash(), logsQuery)
+          .forEach(logWithMetadata -> sendLogToSubscription(logWithMetadata, subscription));
```

Comment:
I understand that we might not be able to reach the nirvana of not checking anything on disk for each block added event like we can with the mainnet side but could we at least take advantage of the fact that we already have the logs from the block added event and get the bare minimum from the privacy storage to filter a subset of them?

Reply:
I don't think I understand your suggestion. The logs in the block added event are only logs generated by public contracts. We can't really use them to find private logs.

Reply:
Ahh, my misunderstanding then.

Reply:
Sem problemas! ;)